tutorials/plot_stats_cluster_1samp_test_time_frequency.py

```diff
@@ -72,17 +73,20 @@
 # evoked_data = np.mean(data, 0)
 
 # Factor to down-sample the temporal dimension of the PSD computed by
-# single_trial_power.  Decimation occurs after frequency decomposition and can
+# tfr_transform.  Decimation occurs after frequency decomposition and can
 # be used to reduce memory usage (and possibly computational time of downstream
 # operations such as nonparametric statistics) if you don't need high
 # spectrotemporal resolution.
 decim = 5
 frequencies = np.arange(8, 40, 2)  # define frequencies of interest
 sfreq = raw.info['sfreq']  # sampling in Hz
-epochs_power = single_trial_power(data, sfreq=sfreq, frequencies=frequencies,
-                                  n_cycles=4, n_jobs=1,
-                                  baseline=(-100, 0), times=times,
-                                  baseline_mode='ratio', decim=decim)
+epochs_power = tfr_transform(data, frequencies, sfreq=sfreq, n_cycles=4.,
```

Comment:
I can make `tfr_transform` private if you really want, but I'll let you take care of the refactoring of high level objects that detect the input type, pick the right channels etc. The main annoyance for me is that the function you're thinking of returns an MNE object (currently, `AverageTFR`, or tuple of `AverageTFR`, and we'd need to create a new class for single trials etc, deal with 'complex' versus 'power' or  'inter-trial coherence' output types etc).

I'm personally against these these kind of heavy object approaches. The people I work with generally complain that most package enforce using complex structures to be able to use their functions, and therefore just re-code them directly. IMO, time frequency transform should be usable with numpy arrays: we don't need to know whether it's an MEG or an EEG channels, it's layout etc, many users just want a computationally efficient convolution.